drivers: Modification of IFX ADC driver to function with new platform

mcatee-infineon opened the pull request:
- Modification of IFX drivers for ADC functionality on newly integrated platform, cyw920829m2evk_02
	- Addition of configuration files to relevant sample/test for cyw920829m2evk_02

ifyall commented on drivers/adc/adc_ifx_cat1.c:
```diff
@@ -33,15 +35,24 @@ LOG_MODULE_REGISTER(ifx_cat1_adc, CONFIG_ADC_LOG_LEVEL);
 #define ADC_CAT1_REF_INTERNAL_MV        (1200u)
 
+#if defined(CONFIG_SOC_FAMILY_INFINEON_CAT1B)
+#define IFX_ADC_NUM_CHANNELS 12
```
Where does the number '12' come from and can it change for new CAT1B devices in the future? Ideally, rather than assigning IFX_ADC_NUM_CHANNELS a hardcoded value, we should get that value from the appropriate device header file or via DTSI.

mcatee-infineon replied:
This is a fair point and should be addressed now.

anangl commented on tests/drivers/adc/adc_api/boards/cyw920829m2evk_02.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_TEST_USERSPACE=n
```
Why?

mcatee-infineon replied:
This was originally done to avoid a fault that was occurring when running in userspace. This has now been addressed in this PR with modifications to mpu.

gmarull reviewed (dismissed):
code contains unrelated formatting changes, also, please, split into independent commits (drivers, sample changes, etc.)

mcatee-infineon committed:
soc: modifications for cyw920829m2evk_02 mpu regions

	- Modifcation to mpu information, in relation to functionality
	  in userspace

Signed-off-by: McAtee Maxwell <maxwell.mcatee@infineon.com>

mcatee-infineon committed:
adc: modifications to support adc on cyw920829m2evk_02 platform

	- Modifications to adc driver
	- Modifications to clock_control driver
	- Add adc to board's yaml
	- Add adc to relevant dts file

Signed-off-by: McAtee Maxwell <maxwell.mcatee@infineon.com>

mcatee-infineon committed:
adc: add cyw920829m2evk_02 overlays to relevant adc samples/tests

	- Add overlay for samples/.../adc_dt
	- Add overlay for tests/.../adc_api

Signed-off-by: McAtee Maxwell <maxwell.mcatee@infineon.com>

mcatee-infineon commented:
> code contains unrelated formatting changes, also, please, split into independent commits (drivers, sample changes, etc.)

@gmarull the changes you requested should be up now.
